crypto/s2n_fips.c

```diff
@@ -16,26 +16,18 @@
 #include <openssl/crypto.h>
 
 #include "crypto/s2n_fips.h"
+#include "crypto/s2n_libcrypto.h"
 
 static int s2n_fips_mode = 0;
 
 int s2n_fips_init(void)
 {
     s2n_fips_mode = 0;
 
-    /* FIPS mode can be checked if OpenSSL was configured and built for FIPS which then defines OPENSSL_FIPS.
-     *
-     * AWS-LC always defines FIPS_mode() that you can call and check what the library was built with. It does not define
-     * a public OPENSSL_FIPS/AWSLC_FIPS macro that we can (or need to) check here
-     *
-     * Note: FIPS_mode() does not change the FIPS state of libcrypto. This only returns the current state. Applications
```

Comment:
This comment feels important- does it live elsewhere? (not seeing it in USAGE.md)

Reply:
But this necessarily can't be true for interning. In this case, s2n has to handle initialising the fips module - somehow - if required.

Reply:
reverted fips changes